micronota/parsers/gff.py

```diff
@@ -0,0 +1,297 @@
+documentation.
+
+Derives_from
+Used to disambiguate the relationship between one feature and another when the
+relationship is a temporal one rather than a purely structural "part of" one.
+This is needed for polycistronic genes.
+
+Note
+A free text note.
+
+Dbxref
+A database cross reference. See the GFF3 specification for more information.
+
+Ontology_term
+A cross reference to an ontology term.
+
+Multiple attributes of the same type are indicated by separating
+the values with the comma ","
+
+
+Format Support
+--------------
+**Has Sniffer: Yes**
+
++------+------+---------------------------------------------------------------+
+|Reader|Writer|                          Object Class                         |
++======+======+===============================================================+
+|Yes   |Yes   |:mod:`skbio.metadata.IntervalMetadata` objects                 |
++------+------+---------------------------------------------------------------+
+|Yes   |No    |generator of :mod:`skbio.metadata.IntervalMetadata` objects    |
++------+------+---------------------------------------------------------------+
+
+
+Reference
+---------
+.. [#] http://gmod.org/wiki/GFF3
+'''
+
+from skbio.io import create_format, FileFormatError
+from skbio.metadata import IntervalMetadata, Feature
+from skbio.io.format._base import (
+    _line_generator, _too_many_blanks)
+from skbio.io.format._base import _get_nth_sequence as _get_nth_record
+
+
+class GFFFormatError(FileFormatError):
+    pass
+
+gff = create_format('gff')
+
+# Annotation headers
+_ANNOTATION_HEADERS = [
+          'SEQID',
+          'SOURCE',
+          'TYPE',
+          'SCORE',
+          'STRAND',
+          'PHASE',
+          'ATTR'
+          ]
+
+_GFF_HEADERS = [
+          'SEQID',
+          'SOURCE',
+          'TYPE',
+          'START',
+          'END',
+          'SCORE',
+          'STRAND',
+          'PHASE',
+          'ATTR'
+          ]
+
+# Attribute headers
+_ATTR_HEADERS = [
+        'ID',
+        'Name',
+        'Alias',
+        'Parent',
+        'Target',
+        'Gap',
+        'Derives_from',
+        'Note',
+        'Dbxref',
+        'Ontology_term'
+        ]
+
+
+@gff.sniffer()
+def _gff_sniffer(fh):
+    # check the 1st real line is a valid ID line
+    if _too_many_blanks(fh, 5):
+        return False, {}
+
+    try:
+        line = next(_line_generator(fh, skip_blanks=True, strip=False))
+    except StopIteration:
+        return False, {}
+
+    if line.startswith('##gff-version 3'):
+        return True, {}
+    else:
+        return False, {}
+
+
+def _is_float(input):
+    try:
+        float(input)
+    except ValueError:
+        return False
+    return True
+
+
+def _parse_attr(s):
+    _field = s.split(';')
+    _attr = ()
+    for f in _field:
+        try:
+            _type, _val = f.split('=')
+        except ValueError:
+            _type, _val = 0, 0
+
+        _attr = _attr + (_type, _val)
```

Comment:
`tuple` is not good for mutation operation - recommend replacing it with `list`. and in terms of data structure, it is probably better to return a list of (key, value) than `[key1, val1, key2, val2, ...]` (if I understand you code correctly)?